libs/pageserver_api/src/models.rs

```diff
@@ -54,6 +54,9 @@ use bytes::{BufMut, Bytes, BytesMut};
 )]
 #[serde(tag = "slug", content = "data")]
 pub enum TenantState {
+    /// This tenant is not yet loaded
+    ///
+    NotLoaded,
```

Comment:
Do I understand correctly that we don't in fact ever use this internally, only map the Option::<Arc<Tenant>>::None to this value with the `map_or`?

If so, we should note that here. I'd really want this conversion to only happen at the last minute when describing the JSON, but it's a shame we have the `pageserver_api` crate.. We could probably still do it with some light serde magic.

Reply:
> Do I understand correctly that we don't in fact ever use this internally, only map the Option::<Arc>::None to this value with the map_or?

Yes, it is true. Not loaded tenants internally handled using `OnceSet` 
`NotLoaded` state is needed just to report state of such tenants in `list_tenants`